Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -1179,6 +1186,220 @@ private bool IsWithinScheduledServerResetTimes()
             return result;
         }
 
+        private DateTime GetBrokerTime()
+        {
+            return DateTime.Now.Add(_brokerTimeDiff);
+        }
+        void HandleBrokerTime(object sender, IB.CurrentTimeEventArgs e)
+        {
+            DateTime brokerTime = e.Time.ToLocalTime();
+            _brokerTimeDiff = brokerTime.Subtract(DateTime.Now);
+        }
+        TimeSpan _brokerTimeDiff = new TimeSpan(0);
+
+
+        /// <summary>
+        /// IDataQueueHandler interface implementaion 
+        /// </summary>
+        /// 
+        public IEnumerable<Data.BaseData> GetNextTicks()
+        {
+            var ticks = new List<Tick>();
+            Tick tick;
+
+            while (_ticks.TryDequeue(out tick))
```

Comment:
we may want to limit the maximum number of ticks being dequeued to 100 or 1000, just so we can gaurantee that this function actually returns, I'm considering the case where we're able to push ticks in at a similar rate to them being pulled off, possibly occurring from subscribing to a myriad of symbols.